File: src/main/java/com/speedscale/newboots/InventoryInitializer.java
package com.speedscale.newboots;

import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;
import org.springframework.beans.factory.annotation.Autowired;
import java.util.*;

@Component
public class InventoryInitializer implements CommandLineRunner {
    @Autowired
    private InventoryRepository inventoryRepository;

    @Override
    public void run(String... args) {
        if (inventoryRepository.count() == 0) {
            List<Inventory> initialInventory = new ArrayList<>();
            initialInventory.add(new Inventory("journal", 25, Map.of("h", 14, "w", 21, "uom", "cm"), "A"));
            initialInventory.add(new Inventory("notebook", 50, Map.of("h", 8.5, "w", 11, "uom", "in"), "A"));
            initialInventory.add(new Inventory("paper", 100, Map.of("h", 8.5, "w", 11, "uom", "in"), "D"));
            initialInventory.add(new Inventory("planner", 75, Map.of("h", 22.85, "w", 30, "uom", "cm"), "D"));
            initialInventory.add(new Inventory("postcard", 45, Map.of("h", 10, "w", 15.25, "uom", "cm"), "A"));
            inventoryRepository.saveAll(initialInventory);
        }
    }
} 

File: src/main/kotlin/com/speedscale/newboots/InventoryInitializer.kt
package com.speedscale.newboots

import org.bson.Document
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class InventoryInitializer(private val inventoryService: InventoryService) : CommandLineRunner {

    override fun run(vararg args: String?) {
        if (inventoryService.count() == 0L) {
            val initialInventory = listOf(
                Inventory(
                    item = "journal",
                    qty = 25,
                    size = Document(mapOf("h" to 14, "w" to 21, "uom" to "cm")),
                    status = "A"
                ),
                Inventory(
                    item = "notebook",
                    qty = 50,
                    size = Document(mapOf("h" to 8.5, "w" to 11, "uom" to "in")),
                    status = "A"
                ),
                Inventory(
                    item = "paper",
                    qty = 100,
                    size = Document(mapOf("h" to 8.5, "w" to 11, "uom" to "in")),
                    status = "D"
                ),
                Inventory(
                    item = "planner",
                    qty = 75,
                    size = Document(mapOf("h" to 22.85, "w" to 30, "uom" to "cm")),
                    status = "D"
                ),
                Inventory(
                    item = "postcard",
                    qty = 45,
                    size = Document(mapOf("h" to 10, "w" to 15.25, "uom" to "cm")),
                    status = "A"
                )
            )
            inventoryService.saveAll(initialInventory)
        }
    }
}
